dGame/dUtilities/VanityUtilities.cpp

```diff
@@ -28,6 +32,26 @@ void VanityUtilities::SpawnVanity() {
 	}
 
 	const uint32_t zoneID = Game::server->GetZoneID();
+	if (!m_NPCs.empty()){
+		for (auto& npc : m_NPCs) {
+			if (npc.m_ID == LWOOBJID_EMPTY) continue;
+			if (npc.m_LOT == 176){
+				Game::zoneManager->RemoveSpawner(npc.m_ID);
+			} else{
+				auto* entity = Game::entityManager->GetEntity(npc.m_ID);
+				if (!entity) continue;
+				entity->Smash(LWOOBJID_EMPTY, eKillType::VIOLENT);
+			}
+		}
+		m_NPCs.clear();
+	}
+	
+	if (!m_Parties.empty()){
+		m_Parties.clear();
+	}
+	if (!m_PartyPhrases.empty()){
+		m_PartyPhrases.clear();
+	}
```

Comment:
You can remove the if wrapper on all 3 of these, clear already has cases to handle this and the for loop will just do nothing if it is empty.
nit: `const auto& npc`